network/src/peer_manager/mod.rs

```diff
@@ -797,6 +798,7 @@ where
         let peer = Peer::new(
             self.network_context.clone(),
             self.executor.clone(),
+            TimeService::real(),
```

Comment:
How about one TimeService in PeerManager and arc it to all the Peer instances?  I'm assuming the end state would be to have the TimeService piped through all the individual pieces, which would mean it would need to exist in PeerManager anyways?

Reply:
I left out propagating the TimeService through PeerManager for the next PR. Also `TimeService::real` is just an empty struct; there's no cost to creating a new one or cloning it.